src/test/java/core/basesyntax/servise/transaction/impl/BalanceTransactionHandlerTest.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.servise.transaction.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.servise.transaction.TransactionHandler;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BalanceTransactionHandlerTest {
+    private static final int BALANCE_QUANTITY = 100;
+    private static final String ITEM = "item";
+    private static TransactionHandler transactionHandler;
+
+    @BeforeClass
+    public static void init() {
+        transactionHandler = new BalanceTransactionHandler();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void proceed_negativeQuantity_notOk() {
+        transactionHandler.proceedTransaction(ITEM, -BALANCE_QUANTITY);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void proceed_duplicateBalanceForSameItem_notOk() {
+        transactionHandler.proceedTransaction(ITEM, BALANCE_QUANTITY);
+        transactionHandler.proceedTransaction(ITEM, BALANCE_QUANTITY);
```

Comment:
I think here for clean test you need to place initial entry (fruit and quantity) into the storage manually without using  your proceedTransaction method for it. Because this method is under testing, so you cannot be sure its working correctly

Reply:
agree